src/test/scala/uk/gov/nationalarchives/LambdaTest.scala

```diff
@@ -345,7 +346,14 @@ class LambdaTest extends AnyFlatSpec with BeforeAndAfterEach {
       .build()
     override val dynamoClient: DADynamoDBClient[IO] = new DADynamoDBClient[IO](asyncDynamoClient)
     override val s3Client: DAS3Client[IO] = DAS3Client[IO](asyncS3Client)
-    override val ingestDateTime: OffsetDateTime = OffsetDateTime.parse("2023-09-01T00:00Z")
+    override def getXmlCreator: XMLCreator = XMLCreator(OffsetDateTime.parse("2023-09-01T00:00Z"))
+  }
+
+  "getXmlCreator" should "return an XmlCreator with an ingest time within microseconds of the current time" in {
```

Comment:
I'm not sure that we need this test. It's comparing times dynamically which will make this flaky if there's any delay running the test. It's unlikely but possible.

It also doesn't really tell you very much you haven't already tested. There's already a test that checks that the date passed in gets passed to the `TransferDateTime` element. 

You also need a val in the XMLCreator class which is only for this test and isn't used anywhere in the main code. It's best not to expose internal class variables if you don't need to in the main code. 

It looks like the issue with this was that the datetime was that it was initialised outside of the handler function and the lambda will cache the value. You're now calling it inside the handler function which should fix it but this test doesn't cover that.